core/lib/spree/core/engine.rb

```diff
@@ -105,6 +105,12 @@ class Engine < ::Rails::Engine
         ]
       end
 
+      initializer 'spree.promo.register.promotions.shipping_actions', before: :load_config_initializers do |app|
```

Comment:
Thanks for the clarification. I have no good ideas other than the one you proposed. Anyway at the moment I can't figure out a scenario when removing something from that `actions` list would break existing orders and promotions. I mean, isn't the same if we remove another action from that `actions` list, like `Spree::Promotion::Actions::CreateAdjustment`?